### PR TITLE
H2 IstioIngressIdentifier and IstioLogger

### LINKED ISSUE #1480: GRPC in Istio
We still need an h2 IstioLogger, IstioIngressIdentifier, and lots of testing in order to support grpc.

### PR DESCRIPTION
**Pull request by esbie**
These are largely copies of their http counterparts. 
Adds the concept of loggers to the h2 protocol. Fixes #1480 

Validation: Manually tested against hello world grpc.

### CODE REVIEW

File: k8s/src/main/scala/io/buoyant/k8s/istio/IstioLoggerBase.scala [hunk 1]
```diff
@@ -0,0 +1,46 @@
+package io.buoyant.k8s.istio
+
+import com.twitter.finagle._
+import com.twitter.util.Duration
+
+trait IstioLoggerBase {
```
Comment (siggy) [inline k8s/src/main/scala/io/buoyant/k8s/istio/IstioLoggerBase.scala]: nice!

File: linkerd/protocol/h2/src/main/scala/io/buoyant/linkerd/protocol/h2/IstioIngressIdentifier.scala [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+              val dst = Dst.Path(path, baseDtab(), Dtab.local)
+              new IdentifiedRequest(dst, req)
+            }
+          case _ => Future.value(new UnidentifiedRequest(s"ingress path ${ingressPath.svc}:${ingressPath.port} does not match any istio vhosts"))
+        }
+    }
+  }
+
+  def redirectRequest(redir: HTTPRedirect, req: Request): Future[Nothing] = {
+    val resp = Response(Status.Found, Stream.empty(0))
```
Comment (adleong) [inline linkerd/protocol/h2/src/main/scala/io/buoyant/linkerd/protocol/h2/IstioIngressIdentifier.scala]: I can't find where Stream.empty(Int) is defined.

File: linkerd/protocol/h2/src/main/scala/io/buoyant/linkerd/protocol/h2/H2LoggerConfig.scala [hunk 3]
```diff
@@ -0,0 +1,56 @@
+package io.buoyant.linkerd.protocol.h2
+
+import com.fasterxml.jackson.annotation.JsonIgnore
+import com.twitter.finagle.buoyant.h2.{Request, Response}
+import com.twitter.finagle.{Filter, ServiceFactory, Stack, Stackable}
+import com.twitter.finagle.stack.nilStack
+import io.buoyant.config.PolymorphicConfig
+
+abstract class H2LoggerConfig extends PolymorphicConfig { config =>
```
Comment (siggy) [inline linkerd/protocol/h2/src/main/scala/io/buoyant/linkerd/protocol/h2/H2LoggerConfig.scala]: you may have already looked at this, but any thoughts on sharing code with `HttpLoggerConfig`? they're pretty similar, though i imagine the `Request`, `Response` types could be a pain to abstract.

same question for the `IstioLogger` classes.

Comment (esbie) [inline linkerd/protocol/h2/src/main/scala/io/buoyant/linkerd/protocol/h2/H2LoggerConfig.scala]: I looked into it briefly, but let me squint at it some more

File: linkerd/protocol/h2/src/main/scala/io/buoyant/linkerd/protocol/h2/IstioIngressIdentifier.scala (hunk 2 above)
Comment (esbie) [inline linkerd/protocol/h2/src/main/scala/io/buoyant/linkerd/protocol/h2/IstioIngressIdentifier.scala]: Bleh, yeah thanks for looking at this, it's a different Stream. Will update to h2.Stream

File: linkerd/protocol/h2/src/main/scala/io/buoyant/linkerd/protocol/h2/H2LoggerConfig.scala (hunk 3 above)
Comment (esbie) [inline linkerd/protocol/h2/src/main/scala/io/buoyant/linkerd/protocol/h2/H2LoggerConfig.scala]: `HttpLoggerConfig`/`H2LoggerConfig`: `new Stack.Module` itself already does good job of reducing to the essentials.

`IstioLoggerConfig`: I could pull out host, port, client, etc into it's own trait (`IstioLoggerConfig extends HttpConfig with IstioLoggerConfigBase`), but having overrides in all three places was unnecessarily confusing. 


In general I only focus on DRYing things up when there are 3 copies of something, and didn't find a lot of value in sharing code between configs.

**Pull request merged**